include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -424,6 +424,15 @@ class ComponentLoweringStateInterface {
     Block *body = component.getBodyBlock();
     builder.setInsertionPoint(body, body->begin());
     auto name = TLibraryOp::getOperationName().split(".").second;
+    if constexpr (TLibraryOp::template hasTrait<FloatingPoint>()) {
+      auto standard = TLibraryOp::getFloatingPointStandard();
+      if (standard == FloatingPointStandard::IEEE754) {
+        assert(name.consume_front("ieee754.") &&
+               "IEEE754 type operation's name must begin with 'ieee754'");
+        std::string modified = llvm::join_items("", "std_", name, "FN");
+        name = modified;
```

Comment:
To avoid a potential copy, just use the r-value:

```suggestion
        name = llvm::join_items("", "std_", name, "FN");
```

Reply:
I don't think `name = llvm::join_items("", "std_", name, "FN");` work
`llvm::join_items` returns a temporary `std::string` so it cannot directly assign to a `StringRef` (`name` here). So I had to assign it to a variable `modified` before assigning to `name`...

Is there any idiomatic way to deal with this, `name = StringRef(llvm::join_items("", "std_", name, "FN"))`?